dynamodb/src/main/scala/zio/dynamodb/DynamoDBExecutorImpl.scala

```diff
@@ -263,14 +263,32 @@ private[dynamodb] final case class DynamoDBExecutorImpl private (dynamoDb: Dynam
   }
 
   private def doScanAll(scanAll: ScanAll): ZIO[Any, Throwable, Stream[Throwable, Item]] =
-    ZIO.succeed(
-      dynamoDb
-        .scan(generateScanRequest(scanAll))
-        .mapBoth(
-          awsError => awsError.toThrowable,
-          item => toDynamoItem(item)
-        )
-    )
+    if (scanAll.totalSegments > 1) {
```

Comment:
This `if` could probably be removed and we could just have the `foreachPar` branch. We'd end up spawning one more fiber in the non-parallel case (to my understanding) but I think that may be acceptable?